### PR TITLE
Set configuration using untiled operations.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**

### CODE REVIEW

**Review by antiagainst — COMMENTED** [review]
Awesome, thanks!

**Review by ThomasRaoux — APPROVED** [review]
LGTM

**Review by hanhanW — APPROVED** [review]

File: iree/compiler/Codegen/Common/test/convert_to_destination_passing_style.mlir [hunk 1]
```diff
@@ -468,3 +468,62 @@ func @unused_ins_operand() {
...
+// CHECK-LABEL: func @three_init_tensor_uses()
+//       CHECK: %[[OUTPUT:.+]] = hal.interface.binding.subspan set(0) binding(1)
+//   CHECK-NOT:   linalg.init_tensor
+//       CHECK:   %[[LOAD:.+]] = flow.dispatch.tensor.load %[[OUTPUT]]
+//   CHECK-NOT:   linalg.init_tensor
+//       CHECK:   linalg.fill(%{{.+}}, %[[LOAD]])
+//       CHECK:   linalg.generic
+//  CHECK-SAME:       outs(%[[LOAD]] :
+//       CHECK:   linalg.generic
+//  CHECK-SAME:       outs(%[[LOAD]] :
```
Comment (hanhanW) [inline iree/compiler/Codegen/Common/test/convert_to_destination_passing_style.mlir:529]: Should have a new line at the end of the file.

image

File: iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 2]
```diff
@@ -634,68 +673,135 @@ static LogicalResult setRootConfigImpl(
   // Redirect to individual operations.
   auto setRootConfigFn = [&](Operation *op) -> LogicalResult {
     return TypeSwitch<Operation *, LogicalResult>(op)
-        .Case<linalg::Mmt4DOp, linalg::ContractionOpInterface,
-              IREE::LinalgExt::FftOp>([&](auto op) {
+        .Case<IREE::LinalgExt::FftOp, linalg::GenericOp, linalg::Mmt4DOp>(
+            [&](auto op) {
+              return setRootConfig(entryPointFn, op, tiledLoops);
+            })
+        .Case<linalg::ContractionOpInterface>([&](auto op) {
```
Comment (hanhanW) [inline iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp:680]: Why split this into two cases?

Comment (MaheshRavishankar) [reply iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp:680]: There is an ordering in the cases, the first match is tried first, then the second. We want the `mmt4d` to be picked up before `ContractionOp`, which should be picked up before `LinalgOp` defaults.

Comment (hanhanW) [reply iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp:680]: Got you, thanks!

File: iree/compiler/Codegen/LLVMGPU/test/illegal_configuration.mlir [hunk 3]
```diff
@@ -256,61 +256,44 @@ hal.executable private @matmul_tensors {
...
-          linalg.batch_matmul {lowering.config = #config} ins(%7, %8 : memref<1x8x1024xf32, affine_map<(d0, d1, d2)[s0] -> (d0 * 32768 + s0 + d1 * 1024 + d2)>>, memref<1x1024x32xf32, affine_map<(d0, d1, d2)[s0] -> (d0 * 65536 + s0 + d1 * 64 + d2)>>) outs(%9 : memref<1x8x32xf32, affine_map<(d0, d1, d2)[s0] -> (d0 * 2048 + s0 + d1 * 64 + d2)>>)
-        }
-      }
-    }
-    return
-  }
-}
-}
-}
+// This test might not be valid anymore when setting configuration on untiled ops.
```
Comment (julianwa) [inline iree/compiler/Codegen/LLVMGPU/test/illegal_configuration.mlir:259]: Did you intend to resolve this before committing?

Comment (MaheshRavishankar) [reply iree/compiler/Codegen/LLVMGPU/test/illegal_configuration.mlir:259]: No, Kojo and Thomas are going to fix this after the fact. I had a chat with Thomas and there is an unrelated issue here anyway, and is being addressed in  #8456

**Pull request merged**